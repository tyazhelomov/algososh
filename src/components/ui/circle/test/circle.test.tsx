import renderer from 'react-test-renderer';

import { Circle } from '../circle';
import { ElementStates } from '../../../../types/element-states';
import React from 'react';

describe('Тест элемента Circle:', () => {
  it('Circle без буквы', () => {
    const tree = renderer
      .create(<Circle letter={''} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с буквами', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с head', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} head={'head'}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с react-элементом в head', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} head={<Circle letter={'head'}/>} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с tail', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} tail={'tail'}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с react-элементом в tail', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} tail={<Circle letter={'tail'}/>} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с index', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} index={1}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle с пропом isSmall === true', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} isSmall={true}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle в состоянии default', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} state={ElementStates.Default} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle в состоянии changing', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} state={ElementStates.Changing} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Circle в состоянии modified', () => {
    const tree = renderer
      .create(<Circle letter={'circle'} state={ElementStates.Modified} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
});
