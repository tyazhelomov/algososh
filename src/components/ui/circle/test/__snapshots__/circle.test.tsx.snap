// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест элемента Circle: Circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          head
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          tail
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тест элемента Circle: Circle с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Тест элемента Circle: Circle с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тест элемента Circle: Circle с пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      circle
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
